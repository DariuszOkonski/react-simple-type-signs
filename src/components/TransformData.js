import React from 'react';

const TransformData = (props) => {
  return (
    <React.Fragment>
      <input
        onChange={props.change}
        type="text"
        value={props.value}
        placeholder='type something...'
      />
      <button onClick={props.click}>Reset</button>
    </React.Fragment>
  );
}

export default TransformData;