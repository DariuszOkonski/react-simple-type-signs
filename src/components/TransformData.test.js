import React from 'react';
import ReactDOM from 'react-dom';
import TransformData from './TransformData';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('TransformData tests', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<TransformData />);
  });

  afterAll(() => {
    wrapper = '';
  });

  it('TransformData renders without a problem', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TransformData />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('Snapshot matches', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('All tags exists', () => {
    expect(wrapper.find('input[type="text"]').exists()).toBe(true);
    expect(wrapper.find('button').exists()).toBe(true);
  });

});