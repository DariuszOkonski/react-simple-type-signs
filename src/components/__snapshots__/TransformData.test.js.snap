// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransformData tests Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TransformData />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <input
          placeholder="type something..."
          type="text"
        />,
        <button>
          Reset
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": undefined,
          "placeholder": "type something...",
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Reset",
        "type": "button",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <input
            placeholder="type something..."
            type="text"
          />,
          <button>
            Reset
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": undefined,
            "placeholder": "type something...",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Reset",
          "type": "button",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
